src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -806,16 +806,18 @@ private void appendToSELECT(Compiler c,
 
     private Compiler selectCount() {
         Compiler c = new Compiler(descriptor);
-        if (!fields.isEmpty()) {
-            c.getSELECTBuilder().append("SELECT COUNT(");
-            if (distinct) {
-                c.getSELECTBuilder().append("DISTINCT");
-            }
+        c.getSELECTBuilder().append("SELECT COUNT(");
+
+        if (!fields.isEmpty() && distinct) {
+            c.getSELECTBuilder().append("DISTINCT ");
+            appendFieldList(c, false);
+        } else if (fields.stream().count() == 1) {
```

Comment:
Maybe throw an exception for multiple fields without distinct?

Reply:
```suggestion
        } else if (fields.size() == 1) {
```

Reply:
As discussed internally, we now treat a single field like `COUNT(*)` and multiple fields in combination with count and without distinct will throw an exception because we don't want the behaviour to appear that "magical". Fields in parent classes can be removed by calling something like `getQuery().fields().count()`.